File: app/src/main/java/net/bloople/manga/TouchRejectionFrameLayout.java
package net.bloople.manga;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class TouchRejectionFrameLayout extends FrameLayout {
    private int touchRejectionOffset;

    public TouchRejectionFrameLayout(@NonNull Context context) {
        super(context);
    }

    public TouchRejectionFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public TouchRejectionFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        this(context, attrs, defStyleAttr, 0);
    }

    public TouchRejectionFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);

        TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable.TouchRejectionFrameLayout, 0, 0);

        try {
            touchRejectionOffset = a.getDimensionPixelSize(R.styleable.TouchRejectionFrameLayout_touchRejectionOffset,
                0);
        }
        finally {
            a.recycle();
        }
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return shouldRejectEvent(ev);
    }

    @Override
    public boolean onInterceptTouchEvent (MotionEvent ev) {
        return shouldRejectEvent(ev);
    }

    private boolean shouldRejectEvent(MotionEvent ev) {
        return ev.getY() < touchRejectionOffset || ev.getY() > (getHeight() - touchRejectionOffset);
    }
}

File: app/src/main/java/net/bloople/manga/TouchRejectionFrameLayout.kt
package net.bloople.manga

import android.content.Context
import android.widget.FrameLayout
import android.util.AttributeSet
import android.view.MotionEvent

class TouchRejectionFrameLayout @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet?,
    defStyleAttr: Int = 0,
    defStyleRes: Int = 0
) : FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
    private val touchRejectionOffset: Int

    init {
        val a = context.theme.obtainStyledAttributes(attrs, R.styleable.TouchRejectionFrameLayout, 0, 0)

        touchRejectionOffset = try {
            a.getDimensionPixelSize(
                R.styleable.TouchRejectionFrameLayout_touchRejectionOffset,
                0
            )
        }
        finally {
            a.recycle()
        }
    }

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return shouldRejectEvent(ev)
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return shouldRejectEvent(ev)
    }

    private fun shouldRejectEvent(ev: MotionEvent): Boolean {
        return ev.y < touchRejectionOffset || ev.y > height - touchRejectionOffset
    }
}